src/covariance.h

```diff
@@ -111,44 +111,29 @@ matrix<T> get_compound_symmetry_heterogeneous(const vector<T>& theta, int n_visi
   return get_heterogeneous_cov(sd_values, fun);
 }
 
-// Coding of cov_type coming from the R side.
-enum cov_type_code {
-  unstructured_cov = 1,
-  toeplitz_cov = 2,
-  auto_regressive_cov = 3,
-  auto_regressive_heterogeneous_cov = 4,
-  ante_dependence_cov = 5,
-  compound_symmetry_cov = 6,
-  compound_symmetry_heterogeneous_cov = 7
-};
-
 // Creates a new correlation object dynamically.
 template <class T>
-matrix<T> get_covariance_lower_chol(const vector<T>& theta, int n_visits, int cov_type) {
+matrix<T> get_covariance_lower_chol(const vector<T>& theta, int n_visits, const std::string& cov_type) {
```

Comment:
I  wonder if it was a bit more flexible re: tests here to use a copy in the signature rather than a const reference:
```suggestion
matrix<T> get_covariance_lower_chol(const vector<T>& theta, int n_visits, std::string cov_type) {
```

Reply:
Let me know if you'd prefer that I change this. Initially I was just copying whatever consensus seemed to surface when I was searching around for best practices. With your comment, I tried to search for something more official and came across [this recommendation](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const) from the iso cpp guidelines:

> For “in” parameters, pass cheaply-copied types by value and others by reference to `const`

Then the distinction is whether covariance type indicators are "cheaply-copied". They use `int`s as examples of cheaply copied, and `string`s as examples of "expensive" copies, but presumably they're considering strings that are longer than 4 characters. 

In short, it _seems_ like this is unlikely to be problematic and aligns with the recommended practice, but I'd be fine to change it if you'd prefer.

Reply:
thanks Doug, yeah I vaguely remember this kind of rule, I would say in this case it is still cheap so let's change